src/credentials.rs

```diff
@@ -275,65 +313,126 @@ impl AWSCredentialsProvider for IAMRoleCredentialsProvider {
             // fetch creds in order: env, file, IAM
-            let mut env_provider = EnvironmentCredentialsProvider::new();
-            match env_provider.get_credentials() {
-                Ok(creds) => {
-                    //println!("Found creds in env");
-                    self.credentials = AWSCredentials{ key: creds.get_aws_access_key_id().to_string(),
-                        secret: creds.get_aws_secret_key().to_string(), token: "".to_string(),
-                        expires_at: UTC::now() + Duration::seconds(600) };
-                    return &self.credentials;
-                }
-                Err(_) => () //println!("Whiffed on env")
-            }
-
-            let mut file_provider = FileCredentialsProvider::new();
-            match file_provider.get_credentials() {
-                Ok(creds) => {
-                    //println!("Found creds in file");
-                    self.credentials = AWSCredentials{ key: creds.get_aws_access_key_id().to_string(),
-                        secret: creds.get_aws_secret_key().to_string(), token: "".to_string(),
-                        expires_at: UTC::now() + Duration::seconds(600) };
-                    return &self.credentials;
-                }
-                Err(_) => () //println!("Whiffed on file")
-            }
 
-            let mut iam_provider = IAMRoleCredentialsProvider::new();
-            match iam_provider.get_credentials() {
-                Ok(creds) => {
-                    //println!("Found creds via iam");
-                    self.credentials = AWSCredentials{ key: creds.get_aws_access_key_id().to_string(),
-                        secret: creds.get_aws_secret_key().to_string(), token: creds.get_token().to_string(),
-                        expires_at: UTC::now() + Duration::seconds(600) };
-                    return &self.credentials;
-                }
-                Err(_) => panic!("Couldn't find AWS credentials in environment, default credential file location or IAM role.")
+            if let Ok(creds) = EnvironmentCredentialsProvider::new().get_credentials() {
+                //println!("Found creds in env");
+                self.credentials = Some(creds.clone());
+            } else if let Ok(creds) = ProfileCredentialsProvider::new().get_credentials() {             
+                //println!("Found creds in file");
+                self.credentials = Some(creds.clone());             
+            } else if let Ok(creds) = IAMRoleCredentialsProvider::new().get_credentials() {
+                //println!("Found creds via iam");
+                self.credentials = Some(creds.clone());
+            } else {
+               panic!("Couldn't find AWS credentials in environment, default credential file location or IAM role.")
             }
         }
-        return &self.credentials;
+        Ok(self.credentials.as_ref().unwrap())
     }
 }
+
+fn in_ten_minutes() -> DateTime<UTC> {
```

Comment:
Due to poor math on my part, the README says five minutes instead of 10.  Can you update the README to reflect the 10 minute behavior?